R[write to console]: Loading required package: permute

R[write to console]: Loading required package: lattice

R[write to console]: This is vegan 2.6-4

R[write to console]: Welcome to compositions, a package for compositional data analysis.
Find an intro with "? compositions"


R[write to console]: 
Attaching package: ‘compositions’


R[write to console]: The following objects are masked from ‘package:stats’:

    anova, cor, cov, dist, var


R[write to console]: The following object is masked from ‘package:graphics’:

    segments


R[write to console]: The following objects are masked from ‘package:base’:

    %*%, norm, scale, scale.default


R[write to console]: Error in (function (file = if (onefile) "Rplots.pdf" else "Rplot%03d.pdf",  : 
  too many open devices

Traceback (most recent call last):
  File "evaluate.py", line 692, in <module>
    plot_PCOA_multiple("ibd_150_"+str(odds_ratio) + "_" + str(cond_effect_val) + "_" + str(batch_effect_val) + "_iter_" + str(iter), df_l, methods_list, metadata_l, used_var="batchid", output_root= overall_path+"/simulation_data_eval_small_072623/out_"+str(odds_ratio)+"_"+ str(cond_effect_val) + "_" + str(batch_effect_val) + "_iter_" + str(iter)+'/')
  File "evaluate.py", line 73, in plot_PCOA_multiple
    r.pdf(output_root+dataset_name+"_multi_PCOA_both_batch.pdf", len(batch_corrected_df_l)*6, height=12)
  File "/athena/linglab/scratch/chf4012/miniconda3/envs/bc_benchmark/lib/python3.8/site-packages/rpy2/robjects/functions.py", line 208, in __call__
    return (super(SignatureTranslatedFunction, self)
  File "/athena/linglab/scratch/chf4012/miniconda3/envs/bc_benchmark/lib/python3.8/site-packages/rpy2/robjects/functions.py", line 131, in __call__
    res = super(Function, self).__call__(*new_args, **new_kwargs)
  File "/athena/linglab/scratch/chf4012/miniconda3/envs/bc_benchmark/lib/python3.8/site-packages/rpy2/rinterface_lib/conversion.py", line 45, in _
    cdata = function(*args, **kwargs)
  File "/athena/linglab/scratch/chf4012/miniconda3/envs/bc_benchmark/lib/python3.8/site-packages/rpy2/rinterface.py", line 817, in __call__
    raise embedded.RRuntimeError(_rinterface._geterrmessage())
rpy2.rinterface_lib.embedded.RRuntimeError: Error in (function (file = if (onefile) "Rplots.pdf" else "Rplot%03d.pdf",  : 
  too many open devices

